Lib/test/test_netrc.py

```diff
@@ -126,8 +127,44 @@ def test_security(self):
             os.chmod(fn, 0o622)
             self.assertRaises(netrc.NetrcParseError, netrc.netrc)
 
-def test_main():
-    support.run_unittest(NetrcTestCase)
+    def test_file_not_found_in_home(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        with support.EnvironmentVarGuard() as environ:
+            environ.set('HOME', d)
+            self.assertRaises(FileNotFoundError, netrc.netrc)
+
+    def test_file_not_found_explicit(self):
+        self.assertRaises(FileNotFoundError, netrc.netrc,
+                          file='unlikely_netrc')
+
+    def test_home_not_set(self):
```

Comment:
Sorry, I don't understand how this test is related to the case when HOME is not set. Why not use the simple test suggested by @berkerpeksag above?

Reply:
My suggested test won't work, because of the following code in `expanduser`:

```py
        if 'HOME' not in os.environ:
            import pwd
            userhome = pwd.getpwuid(os.getuid()).pw_dir
```

I have a `.netrc` file in my `HOME` directory so the test won't pass:

```py
    def test_foo(self):
        with support.EnvironmentVarGuard() as environ:
            environ.unset('HOME')
            self.assertRaises(FileNotFoundError, netrc.netrc)
```

```
======================================================================
FAIL: test_foo (test.test_netrc.NetrcTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/berker/projects/cpython/master/Lib/test/test_netrc.py", line 34, in test_foo
    self.assertRaises(FileNotFoundError, netrc.netrc)
AssertionError: FileNotFoundError not raised by netrc
```

Reply:
Thanks, now I see that this is the simplest platform-independent test.